athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaMetadataHandler.java

```diff
@@ -0,0 +1,385 @@
+    {
+        logger.info("doListTables: " + request);
+
+        List<TableName> tables = new ArrayList<>();
+        try {
+            Connection client = getOrCreateConn(request);
+            DatabaseMetaData dbMetadata = client.getMetaData();
+            ResultSet table = dbMetadata.getTables(null,request.getSchemaName(),null,TABLE_TYPES);
+            while (table.next()){
+                tables.add(new TableName(table.getString(TABLE_SCHEMA),table.getString(TABLE_NAME)));
+            }
+        }
+        catch (SQLException e) {
+            logger.info("SQL Exception"+ e.getMessage());
+        }
+        return new ListTablesResponse(request.getCatalogName(), tables);
+
+    }
+
+    /**
+     * Used to get definition (field names, types, descriptions, etc...) of a Table.
+     *
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request Provides details on who made the request and which Athena catalog, database, and table they are querying.
+     * @return A GetTableResponse which primarily contains:
+     * 1. An Apache Arrow Schema object describing the table's columns, types, and descriptions.
+     * 2. A Set<String> of partition column names (or empty if the table isn't partitioned).
+     * 3. A TableName object confirming the schema and table name the response is for.
+     * 4. A catalog name corresponding the Athena catalog that was queried.
+     */
+    @Override
+    public GetTableResponse doGetTable(BlockAllocator allocator, GetTableRequest request) throws SQLException {
+        logger.info("doGetTable: " + request.getTableName());
+
+        SchemaBuilder tableSchemaBuilder = SchemaBuilder.newBuilder();
+        Set<String> partitionCols = new HashSet<>();
+
+        try {
+            Connection client = getOrCreateConn(request);
+            DatabaseMetaData dbMetadata = client.getMetaData();
+            TableName name = request.getTableName();
+            ResultSet definition = dbMetadata.getColumns(null,name.getSchemaName(),name.getTableName(), null);
+            while(definition.next()) {
+
+                //TODO: this needs to change as per the schema of the table in Vertica; Need to add other data types here!!
+                //If Integer
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("Integer")) {
+                    tableSchemaBuilder.addIntField(definition.getString("COLUMN_NAME"));
+                }
+                //If VARCHAR
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("Varchar")) {
+                    tableSchemaBuilder.addStringField(definition.getString("COLUMN_NAME"));
+                }
+                //If Numeric
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("NUMERIC")) {
+                    tableSchemaBuilder.addDecimalField(definition.getString("COLUMN_NAME"), 10, 2);
+                }
```

Comment:
you've already got a todo here, I'd add that this should move into a new class, perhaps a VerticaSchemaUtils class.

Reply:
added the other datatypes and moved the functionality to a new class as suggested